services/src/main/java/org/keycloak/authorization/admin/PolicyEvaluationService.java

```diff
@@ -223,17 +211,18 @@ private KeycloakIdentity createIdentity(PolicyEvaluationRequest representation)
                         clientSession = keycloakSession.sessions().createClientSession(realm, clientModel);
                         userSession = keycloakSession.sessions().createUserSession(realm, userModel, userModel.getUsername(), "127.0.0.1", "passwd", false, null, null);
 
-                        UserSessionModel finalUserSession = userSession;
-                        ClientSessionModel finalClientSession = clientSession;
+                        new TokenManager().attachClientSession(userSession, clientSession);
 
-                        for (ProtocolMapperModel mapping : clientModel.getProtocolMappers()) {
-                            KeycloakSessionFactory sessionFactory = keycloakSession.getKeycloakSessionFactory();
-                            ProtocolMapper mapper = (ProtocolMapper)sessionFactory.getProviderFactory(ProtocolMapper.class, mapping.getProtocolMapper());
-
-                            if (mapper != null && (mapper instanceof OIDCAccessTokenMapper)) {
-                                accessToken = ((OIDCAccessTokenMapper)mapper).transformAccessToken(accessToken, mapping, keycloakSession, finalUserSession, finalClientSession);
+                        Set<RoleModel> requestedRoles = new HashSet<>();
+                        for (String roleId : clientSession.getRoles()) {
+                            RoleModel role = realm.getRoleById(roleId);
+                            if (role != null) {
+                                requestedRoles.add(role);
                             }
                         }
+
+
+                        accessToken = new TokenManager().createClientAccessToken(keycloakSession, requestedRoles, realm, clientModel, userModel, userSession, clientSession);
```

Comment:
I've tested this change in particular and it breaks the evaluation tool. Main reason being:
* The client session can not be obtained later during evaluation and gives a NPE
* The protocol mappers for a client are not being used anymore to push additional claims to the token